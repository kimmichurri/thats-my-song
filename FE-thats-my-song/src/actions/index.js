export const isLoading = (boolean) => ({
  type: 'IS_LOADING',
  isLoading: boolean
});

export const setSongs = (songs) => ({  
  type: 'SET_SONGS',
  songs
});

export const setCurrentSong = (song) => ({
  type: 'SET_CURRENT_SONG',
  song
});

export const setCurrentCategory = (category) => ({
  type: 'SET_CURRENT_CATEGORY',
  category
});

export const setFetchError = (message) => ({
  type: 'SET_FETCH_ERROR',
  message
});

export const setPlaylistSongs = (songs) => ({
  type: 'SET_PLAYLIST_SONGS',
  songs
});