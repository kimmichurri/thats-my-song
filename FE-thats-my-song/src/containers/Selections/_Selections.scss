.genre-selection-radio-wrapper {
  display: flex;
  justify-content: center;
}

.genre-selection-labels {
  font-family: $main-font;
  font-size: 2.5rem;
  font-weight: 800;
}

.inputs-wrapper {
  margin: 5%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

label, input {
  cursor: pointer;
}

.spin-me-button-wrapper {
  display: flex;
  align-items: center;
  flex-direction: column;
  margin-bottom: 20%;
}

.please-select-category {
  font-family: $accent-font;
  font-size: 1.5rem;
}

.spin-me-button {
  background-color: black;
  border: none;
  border-radius: 50%;
  color: white;
  font-family: $main-font;
  font-size: 2rem;
  height: 200px;
  margin-top: 3%;
  padding: 10px 40px;
  width: 200px;
  &:focus {
    outline: 2px solid #5F6E8E;
  }
  &:hover {
    background-color: #120c16;
    color: #B4A7BC;
    cursor: pointer;
    box-shadow: 1px 1px 40px #D65E74;
  }
}