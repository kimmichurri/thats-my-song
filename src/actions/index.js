export const isLoading = (boolean) => ({
  type: 'IS_LOADING',
  isLoading: boolean
});