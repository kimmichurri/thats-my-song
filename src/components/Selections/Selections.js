import React from 'react';

export default function Selections() {
  return (
    <div className='genre-selection-radio-wrapper'>
      <label for='90s'>90s</label>
      <input type='radio' name='90s'></input>
      <label for='00s'>00s</label>
      <input type='radio' name='00s'></input>
      <label for='rap'>Rap</label>
      <input type='radio' name='rap'></input>
      <label for='rock'>Rock</label>
      <input type='radio' name='rock'></input>
      <label for='pop'>Pop</label>
      <input type='radio' name='pop'></input>
      <label for='wildcard'>Wildcard</label>
      <input type='radio' name='wildcard'></input>            
    </div>
  )
}