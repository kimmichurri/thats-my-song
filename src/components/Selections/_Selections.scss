.genre-selection-radio-wrapper {
  display: flex;
  justify-content: center;
}

.genre-selection-labels {
  font-family: $main-font;
  font-size: 2.5rem;
  font-weight: 800;
}

.inputs-wrapper {
  margin: 5%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

label, input {
  cursor: pointer;
}