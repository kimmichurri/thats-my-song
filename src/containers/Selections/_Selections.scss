.genre-selection-radio-wrapper {
  display: flex;
  justify-content: center;
}

.genre-selection-labels {
  font-family: $main-font;
  font-size: 2.5rem;
  font-weight: 800;
}

.inputs-wrapper {
  margin: 5%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

label, input {
  cursor: pointer;
}

.spin-me-button-wrapper {
  display: flex;
  justify-content: center;
  margin-bottom: 5%;
}

.spin-me-button {
  background-color: black;
  border: none;
  border-radius: 50%;
  color: white;
  font-family: $main-font;
  font-size: 2rem;
  height: 200px;
  margin-top: 3%;
  padding: 10px 40px;
  width: 200px;
  &:focus {
    outline: 2px solid #5F6E8E;
  }
}