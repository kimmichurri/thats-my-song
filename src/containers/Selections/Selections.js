import React, { Component } from 'react';
import { fetchSongs } from '../../thunks/fetchSongs';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { setCurrentSong, setCurrentCategory} from '../../actions/index';

class  Selections extends Component {

  setUrl = (e) => {
    let url;
    const { value } = e.target;
    this.props.setCurrentCategory(value);
    switch (value) {
      case 'Wildcard':
        url = `http://voiceboxpdx.com/api/v1/songs/roulette`
        console.log('wildcard', url);
        this.props.fetchSongs(url)
        break;
      default: 
        url = `http://voiceboxpdx.com/api/v1/songs/roulette?tag=${value}`
        console.log(url);
        this.props.fetchSongs(url)
    }
  }

  pickRandomSong = () => {
    let randomIndex = Math.floor(Math.random() * 10); 
    let randomSong = this.props.songs[randomIndex]
    this.props.setCurrentSong(randomSong);
  }

  render() {
    return (
      <section>
        <div className='genre-selection-radio-wrapper'>
          <div className='inputs-wrapper'>
            <label htmlFor='90s' className='genre-selection-labels'>90s</label>
            <input type='radio' name='radio' id='90s' value='90s' onClick={this.setUrl}></input>
          </div>
          <div className='inputs-wrapper'>
            <label htmlFor='00s' className='genre-selection-labels'>00s</label>
            <input type='radio' name='radio' id='00s' value='00s' onClick={this.setUrl}></input>
          </div>
          <div className='inputs-wrapper'>
            <label htmlFor='rap' className='genre-selection-labels'>Rap</label>
            <input type='radio' name='radio' id='rap' value='Rap' onClick={this.setUrl}></input>
          </div>
          <div className='inputs-wrapper'>
            <label htmlFor='rock' className='genre-selection-labels'>Rock</label>
            <input type='radio' name='radio' id='rock' value='Rock' onClick={this.setUrl}></input>
          </div>
          <div className='inputs-wrapper'>
            <label htmlFor='pop' className='genre-selection-labels'>Pop</label>
            <input type='radio' name='radio' id='pop' value='Pop' onClick={this.setUrl}></input>
          </div>
          <div className='inputs-wrapper'>
            <label htmlFor='wildcard' className='genre-selection-labels'>Wildcard</label>
            <input type='radio' name='radio' id='wildcard' value='Wildcard' onClick={this.setUrl}></input>
          </div>
        </div>
        <div className='spin-me-button-wrapper'>
          <button onClick={this.pickRandomSong} className='spin-me-button'>Take me to my tune!</button>
        </div>
      </section>
    )
  }
}

Selections.propTypes = {
  songs: PropTypes.array.isRequired,
  fetchSongs: PropTypes.func.isRequired,
  setCurrentCategory: PropTypes.func.isRequired,
  setCurrentSong: PropTypes.func.isRequired
}

export const mapStateToProps = (state) => ({
  songs: state.setSongs
});

export const mapDispatchToProps = (dispatch) => ({
  fetchSongs: (url) =>  dispatch(fetchSongs(url)),
  setCurrentCategory: (category) => dispatch(setCurrentCategory(category)),
  setCurrentSong: (song) => dispatch(setCurrentSong(song))
});

export default connect(mapStateToProps, mapDispatchToProps)(Selections);

