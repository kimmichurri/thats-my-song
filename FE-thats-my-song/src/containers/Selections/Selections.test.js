import React from 'react';
import { shallow } from 'enzyme';
import { Selections, mapStateToProps, mapDispatchToProps } from './Selections';
jest.mock('../../thunks/fetchSongs');
import { setCurrentCategory, setCurrentSong } from '../../actions';

describe('Selections', () => {
  let wrapper;
  const mockProps = {
    songs: [{title: 'a song', artist: 'kim', id: 1}],
    currentCategory: 'Rap',
    fetchSongs: jest.fn(),
    setCurrentCategory: jest.fn(),
    setCurrentSong: jest.fn()
  }

  beforeEach(() => {
    wrapper = shallow(
      <Selections 
        songs={mockProps.songs}
        currentCategory={mockProps.currentCategory}
        fetchSongs={mockProps.fetchSongs}
        setCurrentCategory={mockProps.setCurrentCategory}
        setCurrentSong={mockProps.setCurrentSong}
      />
    )
  });

  it('should match the snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return songs and current category as props', () => {
      const mockState = {
        setSongs: [{song: 'a'}, {song: 'b'}],
        setCurrentCategory: '90s',
        otherThing: 'thing1',
        anotherThing: ['thing2']
      }
      const expected = {
        songs: [{song: 'a'}, {song: 'b'}],
        currentCategory: '90s'
      }
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should return fetchSongs to dispatch', () => {
      const mockDispatch = jest.fn();
      const fetchSongs = jest.fn();
      const actionToDispatch = fetchSongs('www.songs.com');
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.fetchSongs('www.songs.com');
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('should return setCurrentCategory as props to dispatch', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = setCurrentCategory('90s');
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setCurrentCategory('90s');
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('should return setCurrentSong as props to dispatch', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = setCurrentSong({song: 'some song'});
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setCurrentSong({song: 'some song'});
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});