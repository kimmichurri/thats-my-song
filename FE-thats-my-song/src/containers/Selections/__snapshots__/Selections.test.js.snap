// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selections should match the snapshot 1`] = `
"<section>
  <div className=\\"genre-selection-radio-wrapper\\">
    <div className=\\"inputs-wrapper\\">
      <label htmlFor=\\"90s\\" className=\\"genre-selection-labels\\">
        90s
      </label>
      <input type=\\"radio\\" name=\\"radio\\" id=\\"90s\\" value=\\"90s\\" onClick={[Function]} />
    </div>
    <div className=\\"inputs-wrapper\\">
      <label htmlFor=\\"00s\\" className=\\"genre-selection-labels\\">
        00s
      </label>
      <input type=\\"radio\\" name=\\"radio\\" id=\\"00s\\" value=\\"00s\\" onClick={[Function]} />
    </div>
    <div className=\\"inputs-wrapper\\">
      <label htmlFor=\\"rap\\" className=\\"genre-selection-labels\\">
        Rap
      </label>
      <input type=\\"radio\\" name=\\"radio\\" id=\\"rap\\" value=\\"Rap\\" onClick={[Function]} />
    </div>
    <div className=\\"inputs-wrapper\\">
      <label htmlFor=\\"rock\\" className=\\"genre-selection-labels\\">
        Rock
      </label>
      <input type=\\"radio\\" name=\\"radio\\" id=\\"rock\\" value=\\"Rock\\" onClick={[Function]} />
    </div>
    <div className=\\"inputs-wrapper\\">
      <label htmlFor=\\"pop\\" className=\\"genre-selection-labels\\">
        Pop
      </label>
      <input type=\\"radio\\" name=\\"radio\\" id=\\"pop\\" value=\\"Pop\\" onClick={[Function]} />
    </div>
    <div className=\\"inputs-wrapper\\">
      <label htmlFor=\\"wildcard\\" className=\\"genre-selection-labels\\">
        Wildcard
      </label>
      <input type=\\"radio\\" name=\\"radio\\" id=\\"wildcard\\" value=\\"Wildcard\\" onClick={[Function]} />
    </div>
  </div>
  <div className=\\"spin-me-button-wrapper\\">
    <NavLink to=\\"/song-info\\">
      <button disabled={false} onClick={[Function]} className=\\"spin-me-button\\">
        Take me to my 
        Rap
         tune!
      </button>
    </NavLink>
  </div>
</section>"
`;
